x-pack/platform/plugins/shared/entity_manager/server/lib/entities/ilm_policies/history_ilm_policy.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IlmPutLifecycleRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { EntityDefinition } from '@kbn/entities-schema';
+import { generateHistoryILMPolicyId } from '../helpers/generate_component_id';
+
+export const generateEntitiesHistoryILMPolicy = (
```

Comment:
Per Mike's (suggestion)[https://github.com/elastic/security-team/issues/13692#issuecomment-3219047428], is there a way to apply it only to On-prem/ECH?

Reply:
I'm not sure how to create an ILM Policy that is dependent on deployment mode. Are there examples of policies like that?

Reply:
We could take a config driven approach, have a config value which prevents the ILM policy from being created, and then only set that confg value in serverless